postprocess.ts

```diff
@@ -1,41 +1,55 @@
 // The Flat Data postprocessing libraries can be found at https://deno.land/x/flat/mod.ts
 // Replace 'x' with latest library version
-import { readTXT, readJSON, writeJSON, writeTXT } from 'https://deno.land/x/flat/mod.ts';
-import { parseFeed } from "https://deno.land/x/rss/mod.ts";
+import { readTXT, readJSON, writeJSON, writeTXT } from 'https://deno.land/x/flat/mod.ts'
+import { parseFeed } from "https://deno.land/x/rss/mod.ts"
+import * as cheerio from "https://esm.sh/cheerio@1.0.0-rc.12"
 
-try {
-    const xmlfile = "data.xml" //Deno.args[0] // equivalent to writing 'const filename = 'data.xml'. 
-    if (xmlfile) {
-        // Gets contents xmlfile as a string. We need this because parseFeed accepts a string as input. 
-        const xml:string = await readTXT(xmlfile);
-        const parsed_feed = await parseFeed(xml)
-        writeJSON('data.json', parsed_feed)
-    } 
-    // check if data.json exists, then start parsing links
-
-    const data = await readJSON('data.json')
-    // console.log(data.entries)
-    await Promise.all(data.entries.map(async (item: { id: string|URL|Request; title: { value: string; }; }) => {
-// Get links
-    const response = await fetch(item.id);
-    const response_html = await response.text();
-    // Trims everything after the last '-' - thanks OGP
-    const blacklistChars = /[^A-Za-z0-9-()_+&@,\s]/g
-    const itemTitle: string[] = item.title.value.split('-')
-    let fileName = itemTitle?.pop()?.trim().replace(/ /g, '_')
-    fileName = fileName?.replace(blacklistChars, '')
-
-    // Check if file already exists
-    try {
-        const file = await Deno.open(`./cases/${fileName}.html`)
-        Deno.close(file.rid)
-    } catch(e) {
-        if (e instanceof Deno.errors.NotFound) {
-            writeTXT(`./cases/${fileName}.html`, response_html)
-        }
+const DATA_XML = "data.xml"
+const DATA_JSON = "data.json"
+
+const readXML = async (xmlFile = DATA_XML): Promise<string> => {
+  // Gets contents xmlFile as a string. We need this because parseFeed accepts a string as input. 
+  const xml:string = await readTXT(xmlfile);
+  const parsedFeed = await parseFeed(xml)
+  writeJSON(DATA_JSON, parsed_feed)
+  return parsedFeed
```

Comment:
Issue: parseFeed() returns a Feed object while the promise in line 10 expects a string.
Feed type: https://deno.land/x/rss@0.5.7/mod.ts?s=Feed 
We can address this by:
`import { parseFeed, type Feed } from "https://deno.land/x/rss/mod.ts"`
and change line 11 to `Promise<Feed>`

I'm in favour of using Feed instead of casting parsedFeed to a string as using Feed lets us access + map entries later on for line 51 w/ `Promise.all(data.entries.map(writeHTML))`!